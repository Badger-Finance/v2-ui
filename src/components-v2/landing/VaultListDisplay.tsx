import { VaultState } from '@badger-dao/sdk';
import { makeStyles, Typography, useMediaQuery, useTheme } from '@material-ui/core';
import BigNumber from 'bignumber.js';
import { Loader } from 'components/Loader';
import { ETH_DEPLOY } from 'mobx/model/network/eth.network';
import { StoreContext } from 'mobx/store-context';
import { observer } from 'mobx-react-lite';
import React, { useContext } from 'react';
import { BalanceNamespace } from 'web3/config/namespaces';

import NoVaults from './NoVaults';
import VaultList from './VaultList';
import VaultListItem from './VaultListItem';
import VaultListItemMobile from './VaultListItemMobile';

const useStyles = makeStyles((theme) => ({
	messageContainer: {
		paddingTop: theme.spacing(4),
		textAlign: 'center',
	},
}));

const VaultListDisplay = observer(() => {
	const classes = useStyles();
	const isMobile = useMediaQuery(useTheme().breakpoints.down('sm'));
	const store = useContext(StoreContext);
	const {
		vaults,
		network: { network },
		user,
	} = store;
	const vaultOrder = vaults.getVaultOrder();

	if (vaultOrder === undefined || vaults.vaultsDefinitions === undefined) {
		return <Loader message={`Loading ${network.name} Setts...`} />;
	}

	if (vaultOrder === null || vaults.vaultsDefinitions === null) {
		return (
			<div className={classes.messageContainer}>
				<Typography variant="h4">There was an issue loading setts. Try refreshing.</Typography>
			</div>
		);
	}

	const settListItems = vaultOrder.flatMap((vault) => {
		const badgerVault = vaults.getVaultDefinition(vault);

		if (!badgerVault) {
			return [];
		}

		const scalar = new BigNumber(vault.pricePerFullShare);
		const depositBalance = user.getBalance(BalanceNamespace.Vault, badgerVault).scale(scalar, true);
		const hasNoBalance = depositBalance.tokenBalance.eq(0);

		// Hide the remBadger vault from users who do not have rembadger (this default hides the sett)
		if (badgerVault.vaultToken.address === ETH_DEPLOY.sett_system.vaults['native.rembadger'] && hasNoBalance) {
			return [];
		}

		// Hide deprecated vaults that the user is not deposited into
		if (vault.state === VaultState.Discontinued && hasNoBalance) {
			return [];
		}

		if (isMobile) {
			return <VaultListItemMobile key={vault.vaultToken} vault={vault} />;
		}

		return <VaultListItem vault={vault} key={vault.vaultToken} />;
	});

	if (settListItems.length === 0 && vaults.vaultsFiltersCount === 0) {
		return <NoVaults network={network.name} />;
	}

	return <VaultList settList={settListItems} />;
});

export default VaultListDisplay;
