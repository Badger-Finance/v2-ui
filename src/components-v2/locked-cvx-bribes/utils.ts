import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
