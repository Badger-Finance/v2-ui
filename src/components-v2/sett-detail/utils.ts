import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
