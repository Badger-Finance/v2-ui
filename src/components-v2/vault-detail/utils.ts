import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
