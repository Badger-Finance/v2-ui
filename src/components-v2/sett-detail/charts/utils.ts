import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
