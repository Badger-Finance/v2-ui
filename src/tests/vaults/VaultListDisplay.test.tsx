import React from 'react';
import store from '../../mobx/RootStore';
import { StoreProvider } from '../../mobx/store-context';
import { VaultDTO, VaultState } from '@badger-dao/sdk';
import { customRender } from '../Utils';
import VaultListDisplay from '../../components-v2/landing/VaultListDisplay';
import UserStore from '../../mobx/stores/UserStore';
import { TokenBalance } from '../../mobx/model/tokens/token-balance';
import BigNumber from 'bignumber.js';
import { SAMPLE_EXCHANGES_RATES, SAMPLE_VAULTS } from '../utils/samples';
import VaultStore from '../../mobx/stores/VaultStore';

const mockVaultsInformation = (vaults: VaultDTO[]) => {
	jest.spyOn(VaultStore.prototype, 'vaultsDefinitions', 'get').mockReturnValue(
		new Map(
			vaults.map((vault) => [
				vault.vaultToken,
				{
					depositToken: { address: vault.underlyingToken, decimals: 18 },
					vaultToken: { address: vault.vaultToken, decimals: 18 },
				},
			]),
		),
	);

	store.vaults.getVaultMap = jest
		.fn()
		.mockReturnValue(Object.fromEntries(vaults.map((vault) => [vault.vaultToken, vault])));
};

describe('VaultListDisplay', () => {
	beforeEach(() => {
		mockVaultsInformation(SAMPLE_VAULTS);
		store.prices.exchangeRates = SAMPLE_EXCHANGES_RATES;
	});

	afterEach(() => {
		jest.restoreAllMocks();
		store.vaults.clearFilters();
	});

	it('displays empty search message', () => {
		mockVaultsInformation([]);
		jest.spyOn(VaultStore.prototype, 'vaultsFiltersCount', 'get').mockReturnValue(1);

		const { container } = customRender(
			<StoreProvider value={store}>
				<VaultListDisplay />
			</StoreProvider>,
		);

		expect(container).toMatchSnapshot();
	});

	it('displays no vaults message', () => {
		mockVaultsInformation([]);
		jest.spyOn(VaultStore.prototype, 'vaultsFiltersCount', 'get').mockReturnValue(0);

		const { container } = customRender(
			<StoreProvider value={store}>
				<VaultListDisplay />
			</StoreProvider>,
		);

		expect(container).toMatchSnapshot();
	});

	it('does not display deprecated vaults with no user balance', () => {
		const vaults = [...SAMPLE_VAULTS].splice(0, 1);
		vaults[0].state = VaultState.Deprecated;

		mockVaultsInformation(vaults);

		const { container } = customRender(
			<StoreProvider value={store}>
				<VaultListDisplay />
			</StoreProvider>,
		);

		expect(container).toMatchSnapshot();
	});

	it('uses default sort criteria by default', () => {
		const vaults = [...SAMPLE_VAULTS];

		mockVaultsInformation(vaults);

		jest.spyOn(UserStore.prototype, 'getTokenBalance').mockImplementation((address: string) => {
			if (address === vaults[2].vaultToken) {
				return new TokenBalance(
					{
						address,
						symbol: '',
						decimals: 18,
						name: '',
					},
					new BigNumber(10),
					new BigNumber(2),
				);
			}

			if (address === vaults[1].underlyingToken) {
				return new TokenBalance(
					{
						address,
						symbol: '',
						decimals: 18,
						name: '',
					},
					new BigNumber(1),
					new BigNumber(2),
				);
			}

			return new TokenBalance(
				{
					address,
					symbol: '',
					decimals: 18,
					name: '',
				},
				new BigNumber(0),
				new BigNumber(0),
			);
		});

		const { container } = customRender(
			<StoreProvider value={store}>
				<VaultListDisplay />
			</StoreProvider>,
		);

		expect(container).toMatchSnapshot();
	});
});
