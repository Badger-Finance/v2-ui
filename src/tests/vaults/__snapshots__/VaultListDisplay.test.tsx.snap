// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultListDisplay displays empty search message 1`] = `
<div>
  <div
    class="MuiPaper-root-116 MuiGrid-root-13 makeStyles-root-9 MuiGrid-container-14 MuiGrid-item-15 MuiPaper-elevation1-120 MuiPaper-rounded-117"
  >
    <div
      class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-2-37"
    >
      <div
        class="MuiGrid-root-13 makeStyles-spacingItem-11 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
      />
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-grid-xs-true-47 MuiGrid-grid-lg-4-93"
      >
        <p
          class="MuiTypography-root-144 makeStyles-title-2 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          Vault
        </p>
        <button
          aria-label="sort descending by name"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-242"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          APY
        </p>
        <button
          aria-label="sort descending by APR"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-242"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          My Deposits
        </p>
        <button
          aria-label="sort descending by balance"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-242"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 makeStyles-tvlColumn-8 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          TVL
        </p>
        <button
          aria-label="sort descending by TVL"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-242"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-116 MuiCard-root-189 MuiGrid-root-13 makeStyles-root-186 MuiGrid-container-14 MuiGrid-justify-content-xs-center-31 MuiPaper-elevation1-120 MuiPaper-rounded-117"
  >
    <div
      class="MuiGrid-root-13 makeStyles-content-188 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-grid-xs-10-57 MuiGrid-grid-sm-6-67 MuiGrid-grid-md-3-78"
    >
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-1-36 MuiGrid-direction-xs-column-17 MuiGrid-justify-content-xs-center-31"
      >
        <div
          class="MuiGrid-root-13 MuiGrid-item-15"
        >
          <img
            alt="vault search is empty"
            src="/assets/icons/empty-vaults-search.svg"
          />
        </div>
        <div
          class="MuiGrid-root-13 MuiGrid-item-15"
        >
          <h6
            class="MuiTypography-root-144 MuiTypography-h6-154 MuiTypography-colorTextSecondary-170"
          >
            Hmmm...
          </h6>
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-justify-content-xs-center-31"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145"
        >
          We couldn't find any matches for your selected filters. Try other filters?
        </p>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14"
      >
        <div
          class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-12-59"
        >
          <button
            class="MuiButtonBase-root-183 MuiButton-root-190 MuiButton-contained-198 makeStyles-filtersButton-187 MuiButton-containedPrimary-199"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-191"
            >
              Filters
            </span>
            <span
              class="MuiTouchRipple-root-242"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-12-59"
        >
          <button
            class="MuiButtonBase-root-183 MuiButton-root-190 MuiButton-text-192 MuiButton-textPrimary-193"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-191"
            >
              Go Back To All Vaults
            </span>
            <span
              class="MuiTouchRipple-root-242"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`VaultListDisplay displays no vaults message 1`] = `
<div>
  <div
    class="MuiGrid-root-7 makeStyles-messageContainer-2 MuiGrid-container-8 MuiGrid-direction-xs-column-11"
  >
    <div
      class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-justify-content-xs-center-25"
    >
      <img
        alt="Badger Builder"
        src="/assets/icons/screwdriver-badger.svg"
      />
    </div>
    <div
      class="MuiGrid-root-7 makeStyles-helpTextContainer-5 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-direction-xs-column-11 MuiGrid-justify-content-xs-center-25"
    >
      <h5
        class="MuiTypography-root-110 MuiTypography-h5-119 MuiTypography-colorTextSecondary-136"
      >
        No vaults to display
      </h5>
      <p
        class="MuiTypography-root-110 MuiTypography-body2-111 MuiTypography-colorTextSecondary-136"
      >
        Badgers are working hard to build vaults on 
        Ethereum
        . Switch network for more investment opportunities.
      </p>
    </div>
    <div
      class="MuiGrid-root-7 makeStyles-switchButtonContainer-6 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-justify-content-xs-center-25"
    >
      <button
        class="MuiButtonBase-root-169 MuiButton-root-140 MuiButton-outlined-145 MuiButton-outlinedPrimary-146"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-141"
        >
          Switch Networks
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`VaultListDisplay does not display deprecated vaults with no user balance 1`] = `
<div>
  <div
    class="MuiGrid-root-7 makeStyles-messageContainer-2 MuiGrid-container-8 MuiGrid-direction-xs-column-11"
  >
    <div
      class="MuiGrid-root-7 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-justify-content-xs-center-25"
    >
      <img
        alt="Badger Builder"
        src="/assets/icons/screwdriver-badger.svg"
      />
    </div>
    <div
      class="MuiGrid-root-7 makeStyles-helpTextContainer-5 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-direction-xs-column-11 MuiGrid-justify-content-xs-center-25"
    >
      <h5
        class="MuiTypography-root-110 MuiTypography-h5-119 MuiTypography-colorTextSecondary-136"
      >
        No vaults to display
      </h5>
      <p
        class="MuiTypography-root-110 MuiTypography-body2-111 MuiTypography-colorTextSecondary-136"
      >
        Badgers are working hard to build vaults on 
        Ethereum
        . Switch network for more investment opportunities.
      </p>
    </div>
    <div
      class="MuiGrid-root-7 makeStyles-switchButtonContainer-6 MuiGrid-container-8 MuiGrid-item-9 MuiGrid-justify-content-xs-center-25"
    >
      <button
        class="MuiButtonBase-root-169 MuiButton-root-140 MuiButton-outlined-145 MuiButton-outlinedPrimary-146"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-141"
        >
          Switch Networks
        </span>
        <span
          class="MuiTouchRipple-root-172"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`VaultListDisplay uses default sort criteria by default 1`] = `
<div>
  <div
    class="MuiPaper-root-116 MuiGrid-root-13 makeStyles-root-9 MuiGrid-container-14 MuiGrid-item-15 MuiPaper-elevation1-120 MuiPaper-rounded-117"
  >
    <div
      class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-2-37"
    >
      <div
        class="MuiGrid-root-13 makeStyles-spacingItem-11 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
      />
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-grid-xs-true-47 MuiGrid-grid-lg-4-93"
      >
        <p
          class="MuiTypography-root-144 makeStyles-title-2 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          Vault
        </p>
        <button
          aria-label="sort descending by name"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-280"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          APY
        </p>
        <button
          aria-label="sort descending by APR"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-280"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          My Deposits
        </p>
        <button
          aria-label="sort descending by balance"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-280"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root-13 makeStyles-title-2 makeStyles-columnTitle-7 makeStyles-tvlColumn-8 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-align-items-xs-center-22 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <p
          class="MuiTypography-root-144 MuiTypography-body2-145 MuiTypography-colorTextSecondary-170"
        >
          TVL
        </p>
        <button
          aria-label="sort descending by TVL"
          class="MuiButtonBase-root-183 MuiIconButton-root-174 makeStyles-sortIcon-3 makeStyles-nonSetSort-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-182"
          >
            <img
              alt="sort-icon"
              class="makeStyles-sortIcon-3"
              src="/assets/icons/vaults-sort-icon.svg"
            />
          </span>
          <span
            class="MuiTouchRipple-root-280"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-116 MuiCard-root-195 MuiGrid-root-13 makeStyles-root-186 MuiGrid-container-14 MuiGrid-direction-xs-column-17 MuiPaper-elevation1-120 MuiPaper-rounded-117"
  >
    <div
      class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-2-37 MuiGrid-item-15"
    >
      <div
        class="MuiGrid-root-13 makeStyles-iconBadgeContainer-190 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
      >
        <div
          class="makeStyles-root-196"
        >
          <img
            alt="DIGG logo"
            class="makeStyles-position-197"
            src="/assets/icons/digg.svg"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-true-47 MuiGrid-grid-lg-4-93"
      >
        <h6
          class="MuiTypography-root-144 makeStyles-vaultName-191 MuiTypography-subtitle1-155"
        >
          Digg
        </h6>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiBox-root-201 MuiBox-root-202"
        >
          <p
            class="MuiTypography-root-144 makeStyles-apr-198 MuiTypography-body1-146 MuiTypography-colorTextPrimary-169 MuiTypography-displayInline-172"
          >
            8.17%
          </p>
          <img
            class="makeStyles-apyInfo-200"
            src="/assets/icons/apy-info.svg"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiBox-root-201 MuiBox-root-227"
        >
          <p
            class="MuiTypography-root-144 makeStyles-itemText-188 MuiTypography-body1-146"
          >
            $0.00e-2
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiBox-root-201 MuiBox-root-231"
        >
          <p
            class="MuiTypography-root-144 makeStyles-itemText-188 MuiTypography-body1-146"
          >
            $12,463,112
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-13 makeStyles-tagsContainer-193 MuiGrid-container-14 MuiGrid-spacing-xs-2-37 MuiGrid-item-15 MuiGrid-align-items-xs-center-22"
    >
      <div
        class="MuiGrid-root-13 makeStyles-iconBadgeContainer-190 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
      />
      <div
        class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-2-37"
        >
          <div
            class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
          >
            <span
              class="MuiTypography-root-144 makeStyles-label-233 MuiTypography-caption-147 MuiTypography-displayInline-172"
            >
              Platform:
            </span>
            <div
              class="MuiChip-root-234 makeStyles-tag-232 MuiChip-sizeSmall-235"
            >
              <span
                class="MuiChip-label-256 MuiChip-labelSmall-257"
              >
                Badger
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root-116 MuiCard-root-195 MuiGrid-root-13 makeStyles-root-186 MuiGrid-container-14 MuiGrid-direction-xs-column-17 MuiPaper-elevation1-120 MuiPaper-rounded-117"
  >
    <div
      class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-2-37 MuiGrid-item-15"
    >
      <div
        class="MuiGrid-root-13 makeStyles-iconBadgeContainer-190 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
      >
        <div
          class="makeStyles-root-196"
        >
          <img
            alt="BADGER logo"
            class="makeStyles-position-264"
            src="/assets/icons/badger.svg"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-true-47 MuiGrid-grid-lg-4-93"
      >
        <h6
          class="MuiTypography-root-144 makeStyles-vaultName-191 MuiTypography-subtitle1-155"
        >
          Badger
        </h6>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiBox-root-201 MuiBox-root-265"
        >
          <p
            class="MuiTypography-root-144 makeStyles-apr-198 MuiTypography-body1-146 MuiTypography-colorTextPrimary-169 MuiTypography-displayInline-172"
          >
            8.17%
          </p>
          <img
            class="makeStyles-apyInfo-200"
            src="/assets/icons/apy-info.svg"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiBox-root-201 MuiBox-root-268"
        >
          <p
            class="MuiTypography-root-144 makeStyles-itemText-188 MuiTypography-body1-146"
          >
            $0.00e-2
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-item-15 MuiGrid-justify-content-xs-flex-end-32 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiBox-root-201 MuiBox-root-271"
        >
          <p
            class="MuiTypography-root-144 makeStyles-itemText-188 MuiTypography-body1-146"
          >
            $43,372,000
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-13 makeStyles-tagsContainer-193 MuiGrid-container-14 MuiGrid-spacing-xs-2-37 MuiGrid-item-15 MuiGrid-align-items-xs-center-22"
    >
      <div
        class="MuiGrid-root-13 makeStyles-iconBadgeContainer-190 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
      />
      <div
        class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-true-47"
      >
        <div
          class="MuiGrid-root-13 MuiGrid-container-14 MuiGrid-spacing-xs-2-37"
        >
          <div
            class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
          >
            <span
              class="MuiTypography-root-144 makeStyles-label-233 MuiTypography-caption-147 MuiTypography-displayInline-172"
            >
              Rewards:
            </span>
            <div
              class="MuiChip-root-234 makeStyles-tag-232 MuiChip-sizeSmall-235"
            >
              <span
                class="MuiChip-label-256 MuiChip-labelSmall-257"
              >
                DCA
              </span>
            </div>
          </div>
          <div
            class="MuiGrid-root-13 MuiGrid-item-15 MuiGrid-grid-xs-auto-46"
          >
            <span
              class="MuiTypography-root-144 makeStyles-label-233 MuiTypography-caption-147 MuiTypography-displayInline-172"
            >
              Platform:
            </span>
            <div
              class="MuiChip-root-234 makeStyles-tag-232 MuiChip-sizeSmall-235"
            >
              <span
                class="MuiChip-label-256 MuiChip-labelSmall-257"
              >
                Badger
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
